native/src/hooks/useLocation.ts

```diff
@@ -0,0 +1,141 @@
+import Geolocation, { GeolocationError, GeolocationResponse } from '@react-native-community/geolocation'
+import { useCallback, useEffect, useState } from 'react'
+import { openSettings, RESULTS } from 'react-native-permissions'
+import SystemSetting from 'react-native-system-setting'
+
+import { checkLocationPermission, requestLocationPermission } from '../utils/LocationPermissionManager'
+
+type SuccessfulLocationState = { status: 'ready' }
+type UnavailableLocationState = {
+  status: 'unavailable'
+  message: 'noPermission' | 'notAvailable' | 'timeout' | 'loading'
+}
+
+type LocationStateType = SuccessfulLocationState | UnavailableLocationState
+
+export type LocationType = [number, number]
```

Comment:
there is an interface you could use from geoJSON called Position
Optional to do ;)

Reply:
I prefer being more specific here, as the geojson type only defines the location as [number, number]